app/org/sagebionetworks/bridge/stormpath/StormpathAccount.java

```diff
@@ -73,26 +78,28 @@
         this.studyIdentifier = studyIdentifier;
         this.encryptors = encryptors;
         this.healthIdKey = studyId + HEALTH_CODE_SUFFIX;
-        this.consentSignatureKey = studyId + CONSENT_SIGNATURE_SUFFIX;
-        this.consentSignaturesKey = studyId + CONSENT_SIGNATURES_SUFFIX;
         this.oldHealthIdVersionKey = studyId + OLD_VERSION_SUFFIX;
         this.oldConsentSignatureKey = studyId + CONSENT_SIGNATURE_SUFFIX;
         this.roles = BridgeUtils.convertRolesQuietly(acct.getGroups());
+        this.allSignatures = Maps.newHashMap();
         
-        signatures = decryptJSONFrom(consentSignaturesKey, CONSENT_SIGNATURES_TYPE);
-        if (signatures == null) {
-            signatures = new ArrayList<>();
-        }
-        if (signatures.isEmpty()) {
-            ConsentSignature sig = decryptJSONFrom(consentSignatureKey, ConsentSignature.class);
-            if (sig != null) {
-                signatures.add(sig);
+        for (SubpopulationGuid subpopGuid : subpopGuids) {
+            List<ConsentSignature> signatures = decryptJSONFrom(subpopGuid.getGuid()+CONSENT_SIGNATURES_SUFFIX, CONSENT_SIGNATURES_TYPE);
+            if (signatures == null || signatures.isEmpty()) {
+                ConsentSignature sig = decryptJSONFrom(subpopGuid.getGuid()+CONSENT_SIGNATURE_SUFFIX, ConsentSignature.class);
+                if (sig != null) {
+                    getConsentSignatureHistory(subpopGuid).add(sig);
+                }
+            } else {
+                getConsentSignatureHistory(subpopGuid).addAll(signatures);
```

Comment:
Add unit tests for (a) reading consents for multiple subpops from CustomData and (b) writing consents for multiple subpops into CustomData.